src/events/codetime.ts

```diff
@@ -42,6 +42,10 @@ export class CodeTime implements CodeTimeInterface {
     this.tz_offset_minutes = data.tz_offset_minutes;
   }
 
+  hasData() {
+    return this.keystrokes ? true : false;
```

Comment:
is the this the property that's guaranteed to be there?    are there instances of code time events with just 'lines_deleted'?

maybe we could check `start_time` or `end_time` since this will act as a validation of sorts also.

Reply:
it should always be there as any key event like lines_deleted is a keystroke but i think we may want to check a few properties before determining if it has data or not

Reply:
oh, you said start time or end time, yes, will check that instead